src/3d/qgs3dmapsettings.cpp

```diff
@@ -92,6 +108,7 @@ Qgs3DMapSettings::~Qgs3DMapSettings()
 
 void Qgs3DMapSettings::readXml( const QDomElement &elem, const QgsReadWriteContext &context )
 {
+  disconnectChangedSignalsToSettingsChanged();
```

Comment:
Instead of the connect/disconnect dance here, you could use QgsProjectDirtyBlocker to instead prevent the project from being dirtied during the scope of this function.